guppylang/defs.py

```diff
@@ -34,10 +46,41 @@ class GuppyFunctionDefinition(GuppyDefinition, Generic[P, Out]):
     def __call__(self, *args: P.args, **kwargs: P.kwargs) -> Out:
         return cast(Out, super().__call__(*args, **kwargs))
 
-    def compile(self) -> FuncDefnPointer:
-        from guppylang.decorator import guppy
+    def compile(self) -> Package:
+        """Compile a Guppy function definition to HUGR."""
+        compiled_module = ENGINE.compile(self.id)
+
+        # Look up how many generic params the function has so we can create an empty
+        # partial monomorphization to look up in the context
+        checked_def = ENGINE.checked[self.id]
+        mono_args: PartiallyMonomorphizedArgs | None = None
+        if isinstance(checked_def, MonomorphizableDef):
+            mono_args = tuple(None for _ in checked_def.params)
+
+        _ = ENGINE.compiled[self.id, mono_args]
```

Comment:
If you're not interested in returning a `FuncDefnPointer`, then there is no need to do any of this and the inherited `compile` implementation should be fine